modules/check_in/app/services/v2/chip/service.rb

```diff
@@ -77,14 +77,24 @@ def identifier_params
       def demographic_confirmations
         demographics_needs_update = check_in_body[:demographics_up_to_date] ? false : true
         next_of_kin_needs_update = check_in_body[:next_of_kin_up_to_date] ? false : true
-        {
-          demographicConfirmations: {
-            demographicsNeedsUpdate: demographics_needs_update,
-            demographicsConfirmedAt: Time.zone.now.iso8601,
-            nextOfKinNeedsUpdate: next_of_kin_needs_update,
-            nextOfConfirmedAt: Time.zone.now.iso8601
+        confirmed_at = Time.zone.now.iso8601
+
+        result =
+          {
+            demographicConfirmations: {
+              demographicsNeedsUpdate: demographics_needs_update,
+              demographicsConfirmedAt: confirmed_at,
+              nextOfKinNeedsUpdate: next_of_kin_needs_update,
+              nextOfConfirmedAt: confirmed_at
+            }
           }
-        }
+
+        result.tap do |hash|
+          if Flipper.enabled?(:check_in_experience_chip_service_nok_confirmation_update_enabled)
+            hash[:demographicConfirmations].delete(:nextOfConfirmedAt)
+            hash[:demographicConfirmations].store(:nextOfKinConfirmedAt, confirmed_at)
```

Comment:
Just a thought, wondering if we can add `nextOfConfirmedAt` or `nextOfKinConfirmedAt` based on the feature flag instead of adding `nextOfConfirmedAt` & then deleting and adding `nextOfKinConfirmedAt` key.

Reply:
Good thought! I think this would be the approach if we had implemented this key at the beginning with certain requirements in mind. However, the idea is not to break/touch the existing production hash. If we add the key now based on the condition of the feature toggle, then we would have modified the existing production hash as it's meant to be returned today.

Reply:
Synced offline on the change